api_app/script_analyzers/file_analyzers/unpac_me.py

```diff
@@ -0,0 +1,100 @@
+            )
+        API_KEY_STRING = "Key %s" % self.__api_key
+        self.headers = {"Authorization": API_KEY_STRING}
+        unpac_id = self._upload()
+        logger.info(f"md5 {self.md5} job {self.job_id} uploaded id {unpac_id}")
+        for chance in range(self.max_tries):
+            time.sleep(self.poll_distance)
+            logger.info(
+                f"unpacme polling, try n.{chance + 1}."
+                f" job_id {self.job_id}. starting the query"
+            )
+            status = self._get_status(unpac_id)
+            if status == "fail":
+                logger.error(f"md5 {self.md5} job {self.job_id} analysis has failed")
+                raise AnalyzerRunException("failed analysis")
+            if status != "complete":
+                logger.info(
+                    f"md5 {self.md5} job {self.job_id} id {unpac_id} status {status}"
+                )
+                continue
+            return self._get_report(unpac_id)
+
+    def _req_with_checks(self, url, files=None, post=False):
+        try:
+            if post:
+                r = requests.post(
+                    self.base_url + url, files=files, headers=self.headers
+                )
+            else:
+                headers = self.headers if self.private == "private" else {}
+                r = requests.get(self.base_url + url, files=files, headers=headers)
+            r.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            logger.error(
+                f"md5 {self.md5} job {self.job_id} url {url} has http error {str(e)}"
+            )
+            if post:
+                raise AnalyzerRunException("Monthly quota exceeded!")
+            raise AnalyzerRunException(e)
+        except requests.exceptions.Timeout as e:
+            logger.error(
+                f"md5 {self.md5} job {self.job_id} url {url} has timeout error {str(e)}"
+            )
+            raise AnalyzerRunException(e)
+        except requests.exceptions.RequestException as e:
+            logger.error(
+                f"md5 {self.md5} job {self.job_id} url {url} failed with error {str(e)}"
+            )
+            raise AnalyzerRunException(e)
+        return r
+
+    def _upload(self) -> str:
+        f = open(self.filepath, "rb")
+        file_data = f.read()
+        f.close()
+        files = {"file": (self.filename, file_data)}
+        r = self._req_with_checks("private/upload", files=files, post=True)
+        response = r.json()
+        if "id" not in response:
+            raise AnalyzerRunException(
+                f"md5 {self.md5} job {self.job_id} function upload id not in response"
+            )
+        return response["id"]
+
+    def _get_status(self, unpac_me_id) -> str:
+        response = self._req_with_checks(f"{self.private}/status/{unpac_me_id}")
+        return response.json().get("status", False)
+
+    def _get_report(self, unpac_me_id) -> dict:
```

Comment:
This should be `-> "Dict"` or we can do `from typing import Dict` and then `-> Dict`

Reply:
Why is better to use
 
```
from typing import Dict
```
?
Is not the same at the end?